MyFirstSlideshow/app/src/main/java/com/charles/myfirstslideshow/MainActivity.java

```diff
@@ -10,12 +26,141 @@
+            b = BitmapFactory.decodeStream(c.getInputStream());
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException("Error decoding picture");
+        }
+
+        image1.get().setImageBitmap(b);
+
+        //Update dots
+        if (i % 4 == 0) {
+            dots1[0].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 1) {
+            dots1[1].setBackgroundColor(GREEN);
+            dots1[0].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 2) {
+            dots1[2].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[0].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 3) {
+            dots1[3].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[0].setBackgroundColor(GRAY);
+        }
+    }
+
+    public synchronized void onButtonTwoClicked(View button) {
+        String currentImage = "" + images.get(j % 4);
+        j += 1;
+        // the last url of the array changes every ~ 15 minutes
+        Bitmap b;
+        try {
+            URL url = new URL(currentImage);
+            URLConnection c = url.openConnection();
+            b = BitmapFactory.decodeStream(c.getInputStream());
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException("Error decoding picture");
+        }
+
+        image2.get().setImageBitmap(b);
+
+        //Update dots
+        if (j % 4 == 0) {
+            dots2[0].setBackgroundColor(GREEN);
+            dots2[1].setBackgroundColor(GRAY);
+            dots2[2].setBackgroundColor(GRAY);
+            dots2[3].setBackgroundColor(GRAY);
+        }
+        if (j % 4 == 1) {
+            dots2[1].setBackgroundColor(GREEN);
+            dots2[0].setBackgroundColor(GRAY);
+            dots2[2].setBackgroundColor(GRAY);
+            dots2[3].setBackgroundColor(GRAY);
+        }
+        if (j % 4 == 2) {
+            dots2[2].setBackgroundColor(GREEN);
+            dots2[1].setBackgroundColor(GRAY);
+            dots2[0].setBackgroundColor(GRAY);
+            dots2[3].setBackgroundColor(GRAY);
+        }
+        if (j % 4 == 3) {
+            dots2[3].setBackgroundColor(GREEN);
+            dots2[1].setBackgroundColor(GRAY);
+            dots2[2].setBackgroundColor(GRAY);
+            dots2[0].setBackgroundColor(GRAY);
+        }
+    }
+
+    public void onImageTapped(View image) {
+    }
+    {
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
     }
```

Comment:
This won't work at all. You have an extra set of brackets here. Please remove the first two brackets "{}
```
public void onImageTapped(View image) {
        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);
    }	  
```  
Anyway.  I don't  see a reason why to use "Strict mode" is soo simple app. If you are loading images from an URL you should use libraries which are dedicated to downloading and optimizing images from URL. The most widespread libraries are Piccasso or Glide. Do not handle downloading images on your own.

See one of these libraries and use it. It is simple:
https://github.com/square/picasso
https://github.com/bumptech/glide